plugins/inputs/whois/whois_test.go

```diff
@@ -0,0 +1,198 @@
+			domains:   nil,
+			timeout:   config.Duration(5 * time.Second),
+			expectErr: true,
+		},
+		{
+			name:      "Invalid Timeout (Zero Value)",
+			domains:   []string{"example.com"},
+			timeout:   config.Duration(0),
+			expectErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			plugin := &Whois{
+				Domains: test.domains,
+				Timeout: test.timeout,
+				Server:  test.server,
+				Log:     testutil.Logger{},
+			}
+
+			err := plugin.Init()
+
+			if test.expectErr {
+				require.Error(t, err, "Expected error but got none")
+				return
+			}
+
+			require.NoError(t, err, "Unexpected error during Init()")
+		})
+	}
+}
+
+func TestWhoisCases(t *testing.T) {
+	// Get all directories in testcases
+	folders, err := os.ReadDir("testcases")
+	require.NoError(t, err, "Failed to read testcases directory")
+
+	// Register the WHOIS plugin inside the test
+	inputs.Add("whois", func() telegraf.Input {
+		return &Whois{}
+	})
+
+	// Prepare the influx parser for expectations
+	parser := &influx.Parser{}
+	require.NoError(t, parser.Init())
+
+	for _, f := range folders {
+		if !f.IsDir() {
+			continue
+		}
+
+		testcasePath := filepath.Join("testcases", f.Name())
+		configFilename := filepath.Join(testcasePath, "telegraf.conf")
+		expectedFilename := filepath.Join(testcasePath, "expected.out")
+		expectedErrorFilename := filepath.Join(testcasePath, "expected.err")
+
+		// Compare options for metrics
+		options := []cmp.Option{
+			testutil.IgnoreTime(),
+			testutil.SortMetrics(),
+			testutil.IgnoreFields("expiry"),
+		}
+
+		t.Run(f.Name(), func(t *testing.T) {
+			// Read input data
+			inputData, inputErrors, err := readInputData(testcasePath)
+			require.NoError(t, err)
+
+			// Read expected output
+			var expectedMetrics []telegraf.Metric
+			if _, err := os.Stat(expectedFilename); err == nil {
+				expectedMetrics, err = testutil.ParseMetricsFromFile(expectedFilename, parser)
+				require.NoError(t, err)
+			}
+
+			// Read expected errors
+			var expectedErrors []string
+			if _, err := os.Stat(expectedErrorFilename); err == nil {
+				expectedErrors, err = testutil.ParseLinesFromFile(expectedErrorFilename)
+				require.NoError(t, err)
+			}
+
+			// Load Telegraf plugin config
+			cfg := config.NewConfig()
+			require.NoError(t, cfg.LoadConfig(configFilename))
+			require.Len(t, cfg.Inputs, 1)
+
+			// Get WHOIS plugin instance
+			plugin := cfg.Inputs[0].Input.(*Whois)
+			plugin.whoisLookup = func(_ *whois.Client, domain, _ string) (string, error) {
+				return string(inputData[domain]), inputErrors[domain]
+			}
+			require.NoError(t, plugin.Init())
+
+			var acc testutil.Accumulator
+			require.NoError(t, plugin.Gather(&acc))
+			if len(acc.Errors) > 0 {
+				var actualErrorMsgs []string
+				for _, err := range acc.Errors {
+					actualErrorMsgs = append(actualErrorMsgs, err.Error())
+				}
+				require.ElementsMatch(t, actualErrorMsgs, expectedErrors)
+			}
```

Comment:
I think this must be
```suggestion
			var actualErrorMsgs []string
			for _, err := range acc.Errors {
				actualErrorMsgs = append(actualErrorMsgs, err.Error())
			}
			require.ElementsMatch(t, actualErrorMsgs, expectedErrors)
```

Reply:
Fixed. Thanks